torch/fx/graph.py

```diff
@@ -313,20 +337,26 @@ def format_arg(arg) -> str:
         def format_node(n : Node) -> Optional[str]:
             if n.op == 'placeholder':
                 assert isinstance(n.target, str)
-                placeholder_names.append(n.target)
+                arg_str = n.target
+                arg_str = arg_str + f': {torch.typename(n.type)}' if n.type is not None else ''
+                placeholder_names.append(arg_str)
                 return None
             elif n.op == 'get_attr':
-                return f'%{n.name} : [uses={n.uses}] = self.{n.target}'
+                maybe_typename = f', type={torch.typename(n.type)}' if n.type is not None else ''
+                return f'%{n.name} : [uses={n.uses}{maybe_typename}] = self.{n.target}'
```

Comment:
nit: maybe follow Python's annotation syntax and keep `[]` only for uses?

E.g. `foo : int [uses=2] = 123` instead of `foo : [uses=2, type=int] = 123` now?